sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuColumnarBatchSerializer.scala

```diff
@@ -491,6 +494,120 @@ private class KudoSerializerInstance(
     throw new UnsupportedOperationException
 }
 
+
+
+
+private class KudoGpuSerializerInstance(
+    val metrics: Map[String, GpuMetric],
+    val dataTypes: Array[DataType],
+) extends SerializerInstance {
+  private val serTime = metrics(METRIC_SHUFFLE_SER_STREAM_TIME)
+  private val deserTime = metrics(METRIC_SHUFFLE_DESER_STREAM_TIME)
+
+  override def serializeStream(out: OutputStream): SerializationStream = new SerializationStream {
+
+    override def writeValue[T: ClassTag](value: T): SerializationStream = serTime.ns {
+      val batch = value.asInstanceOf[ColumnarBatch]
+      val numColumns = batch.numCols()
+      withResource(new ArrayBuffer[AutoCloseable](numColumns)) { _ =>
+        if (batch.numCols() > 0) {
+          val firstCol = batch.column(0)
+          firstCol match {
+            case vector: SlicedSerializedColumnVector =>
+              withResource(vector.getWrap) { buf =>
+                val start = vector.getStart
+                val len = vector.getEnd - start
+                val data = buf.slice(start, len)
```

Comment:
maybe simpler to just slice the buffer ahead of time instead of storing offsets explicitly